parlor.go

```diff
@@ -8,37 +8,34 @@ import (
 
 type Parlor struct {
 	suite.Suite
+	testingParlor interface{}
+}
+
+func (p *Parlor) SetTestingParlor(tp TestingParlor) {
+	p.testingParlor = tp
 }
 
 type TestingParlor interface {
 	suite.TestingSuite
-	suite.SetupTestSuite
-	suite.TearDownTestSuite
+	SetTestingParlor(TestingParlor)
 }
 
 func Run(t *testing.T, parlor TestingParlor) {
+	parlor.SetTestingParlor(parlor)
 	suite.Run(t, parlor)
 }
 
-func (p *Parlor) Run(
-	name string,
-	subtest func(),
-	tp TestingParlor,
-) bool {
-	return p.RunWithSetupAndTeardown(
-		name,
-		subtest,
-		tp.SetupTest,
-		tp.TearDownTest,
-	)
-}
+func (p *Parlor) Run(name string, subtest func()) bool {
+	setup := func() {}
+	if i, ok := p.testingParlor.(suite.SetupTestSuite); ok {
```

Comment:
interesting! This works great, but because the `suite.SetupTestSuite` and `suite.TearDownTestSuite` are no longer part of the `parlor.TestingParlor` interface and the interface funcs `SetupTest` and `TearDownTest` are not explicitly called (instead the hidden interface parlor.testingParlor is assertion checked) we lose a functionality of this package:

If the suite you create, and pass to `parlor.Run` does not fulfill the interfaces `suite.SetupTest` and/or `suite.TearDownTest` we no longer get a compilation error. **This change allows your suite to not fulfill those interfaces**

Reply:
We originally created the Parlor package to fix the problem - ensure that teardown functions are called

Is allowing a suite to not declare a `suite.TearDownTest()` and not cause a compile error follow the purpose of this package?

Reply:
I am not sure I agree.  In xUnit style testing, which is what all of this is based off, the system should provide hooks (but not force) the the developer to perform setups and teardowns at varying levels.  The major problem with subtests is that there are no hooks for setups and teardown, which as we have seen in a few places, creates bugs and redundant code.  

To be consistent with the xUnit paradigm, shouldn't parlor give the developer the option (but not force) implementing a subtest setup and teardown and then if they do, run the setup/teardowns for the subtests?

Additional thought, but relatively minor, would it be better to define an interface that provides functions `SetupSubTest` and `TearDownSubTest`? 

... obviously, this is all kind of academic but maybe useful for v2.